sklearn/ensemble/_voting.py

```diff
@@ -335,12 +373,23 @@ def fit(self, X, y, sample_weight=None):
 
             .. versionadded:: 0.18
 
+        **fit_params : dict
+            Parameters to pass to the underlying estimators.
+
+            .. versionadded:: 1.4
+
+                Only available if `enable_metadata_routing=True`,
+                which can be set by using
+                ``sklearn.set_config(enable_metadata_routing=True)``.
+                See :ref:`Metadata Routing User Guide <metadata_routing>` for
+                more details.
+
         Returns
         -------
         self : object
             Returns the instance itself.
         """
-        _raise_for_unsupported_routing(self, "fit", sample_weight=sample_weight)
+        _raise_for_params(fit_params, self, "fit")
```

Comment:
after sample_weight as a positional arg is deprecated, it'll be always in `fit_params`, and we should pop it before this check. Not to be done in this PR, but a comment here or above where sample_weight is deprecated is nice.

Reply:
Okay, got it.

Reply:
Though it's still not fully clear to me. How long do we want to keep backwards compatibility, if 2 releases is not long enough? After the deprecation cycle is over `sample_weight` won't be mentioned in the documentation anymore, so why do we still support it as if it was a usual param?